### ISSUE TITLE
Possibility to generate with the modal token rather than a sample

### ISSUE THREAD

**Issue by SamDuffield** [outlines-dev/outlines#241]
`outlines.text.generate.continuation` generates by sampling the next token from the categorical distribution defined by the logits, however in some cases one may want to just take the mode (highest probability) token.

**Comment by rlouf** [comment]
Why do you want to add greedy search?

**Comment by brandonwillard** [comment]
It should be possible to alter/swap out next-token generating approaches in general.  Perhaps we should start considering how to make that easier first, if it isn't already.

**Comment by SamDuffield** [comment]
> Why do you want to add greedy search?

It might perform better in some cases (I don't have a great feel for this). But it is the default in transformers and I think people will expect this functionality.


> It should be possible to alter/swap out next-token generating approaches in general. Perhaps we should start considering how to make that easier first, if it isn't already.

Swap-ability sounds awesome. Do you have ideas for next-token generation beyond sampling or greedy?

**Comment by SamDuffield** [comment]
In line with [`transformers`](https://github.com/huggingface/transformers/blob/977b2f05d5697f33e51111e4834a127a9a76349f/src/transformers/generation/logits_process.py#L225) we could have a temperature parameter that scales the logits i.e. `logits = logits / temperature` such that `temperature=0` corresponds to greedy sampling

**Comment by blmoistawinde** [comment]
Agree. In many cases we want a definite result for accuracy. Just use the examples in README.

```
prompt = """You are a sentiment-labelling assistant.
Is the following review positive or negative?

Review: This restaurant is just awesome!
"""
answer = generate.choice(model, ["Positive", "Negative"])(prompt)
```

Here we just want "Positive", but sampling will make it possible to get "Negative".

**Comment by rlouf** [comment]
There's a refactor coming soon that allows one to plug in any sampling algorithm (including greedy)
